app/src/main/java/com/yacgroup/yacguide/UpdatableTableActivity.kt

```diff
@@ -110,31 +112,48 @@ abstract class UpdatableTableActivity : TableActivity(), UpdateListener {
 
     protected abstract fun deleteContent()
 
-    protected abstract fun searchRocks(rockName: String): List<Rock>
-
-    private fun _searchRockGlobally() {
-        val searchDialog = Dialog(this)
-        searchDialog.setContentView(R.layout.search_dialog)
-        searchDialog.findViewById<Button>(R.id.searchButton).setOnClickListener {
-            val rockName = searchDialog.findViewById<EditText>(R.id.dialogEditText).text.toString()
-            if (rockName.isEmpty()) {
-                Toast.makeText(searchDialog.context, R.string.hint_no_name, Toast.LENGTH_SHORT).show()
-            } else {
-                val rocks = searchRocks(rockName)
-                if (rocks.isEmpty()) {
-                    Toast.makeText(searchDialog.context, R.string.hint_name_not_found, Toast.LENGTH_SHORT).show()
-                } else {
-                    val rockIds = ArrayList(rocks.map{ it.id })
-                    val intent = Intent(this, SelectedRockActivity::class.java)
-                    intent.putIntegerArrayListExtra(IntentConstants.SELECTED_ROCK_IDS, rockIds)
-                    startActivity(intent)
-                    searchDialog.dismiss()
+    protected abstract fun searchRocks(): List<Rock>
+
+    protected abstract fun searchProjects(): List<Rock>
+
+    protected abstract fun searchBotches(): List<Rock>
+
+    private fun _searchRockGlobally(filter: ClimbingObjectFilter = ClimbingObjectFilter.eNone) {
+        when (filter) {
+            ClimbingObjectFilter.eProject -> _goToFilteredRocksView(searchProjects())
+            ClimbingObjectFilter.eBotch -> _goToFilteredRocksView(searchBotches())
+            else -> {
+                val searchDialog = Dialog(this)
```

Comment:
I think we can use our new `DialogWidgetBuilder` here, too, if you make the text of the positive button configurable. If it is not possible fo some reason, please use at least class`AppCompatDialog` from `androidx.appcompat.app` instead to be consistent within the project.

Reply:
> Isn't the DialogWidgetBuilder restricted to Info + Yes/No action? In that case, I have a more complex dialog containing an EditText field.

Not necessarily, see `WhatsNewInfo` class. 

> Another problem is that AlertDialog is closed by default if clicking on OK. We don't want this here.

OK, this is possible to implement (see `PartnersActivity` class) but at the end it has no advantage over the current solution.